centos_ci.sh

```diff
@@ -46,6 +46,7 @@ function install_core_deps() {
   yum -y install epel-release
   # Get all the deps in
   yum -y install gcc \
+                 golang \
```

Comment:
so where is the version here? How does this ensure that the version does not change? Do we not need to specify one?

Reply:
@hferentschik Here go version comes from official repo of centos which is 1.8.3 currently. This is the reason we are not moving to 1.9

Reply:
The right approach for installing a particular version with YUM is to download the RPM using wget or curl and then use `yum localinstall /path/to/package.rpm`

Reply:
@LalatenduMohanty  big -1 we should never download an rpm locally, assume tomorrow if distribution move with 1.9 then you never know package dependencies which a specific package depend on should be same because it will be break with conflict (you can try to download a lower version of vim rpm locally and then install it, you can see all the conflicts). Also if that was the intentions then why not go for 1.9 because packages are available on Fedora side and we can download but it again breaks on conflict and dependencies resolution.

Reply:
@praveenkumar  It depends on what we want. If we want to hard fix the version at 1.8.3 it is fine as if the distribution moves on and we get conflicts the build will fail eventually and that would be the point when we will have to move to a new version consciously. Personally I think the current code is fine as it is inline with the documentation i.e. >= 1.8.

Reply:
> If we want to hard fix the version at 1.8.3 it is fine using localinstall

If possible we want to pin the version. That's what the existing local install did. I don't understand why we want to open ourself up for potential problems due to the fact the packaged version change.

>  i.e. if the distribution moves on and we get conflicts the build will fail eventually and that would be the point when we will have to move to a new version consciously.

A failing build might be the easy case. There could be much more subtle errors. If we can we should be explicit and imo in this case we can. 

 > Personally I think the current code is fine as it is inline with the documentation i.e. >= 1.8.

For me this change is still a -1. If we can explicitly set the version we should do so.

Reply:
Also, how do I know which version is installed?

Reply:
@hferentschik  We are contradicting our self here because in the documentation we are saying any version >= 1.8 is fine. But we want to pin to a particular version. Lets assume we pinned the version to 1.8.3 and everything is fine. We will never know if there is an issue exists with 1.9 version and then a contributor might use it and get in to issues. Thats why I mentioned that we should decide between = 1.8.3 or >= 1.8.3 in the docs.

Reply:
> We are contradicting our self here because in the documentation we are saying any version >= 1.8 is fine. 

That's local development. Personally, I even use 1.9, but that's a conscious decision. The CI environment is a different beast. The less moving parts the better. 

The main issue is that there is no indication of what version is used whatsoever. I would be ok with the yum install if it were in the form `sudo yum install golang-<version info>`. But then again, we have this function which does a manual install, why change it? Maybe for some reason, we will use a slightly different version from the packaged version in a short time again. Do you then want to dig this code out again. Most likely it is by the forgotten how it was done and someone needs to re-do it.

> But we want to pin to a particular version. Lets assume we pinned the version to 1.8.3 and everything is fine. We will never know if there is an issue exists with 1.9 version and then a contributor might use it and get in to issues. 

You will know since some people will try with newer versions. Or you schedule a dedicated task for the upgrade. If you want to be proactive and get notified early, you could create an additional matrix job which compiles against multiple versions. But this is again an active choice and explicit. In the pull request builder you want to minimise side effects.

> Thats why I mentioned that we should decide between = 1.8.3 or >= 1.8.3 in the docs.

I think I am in this case more inclined to change the docs. Something like "The recommended Golang version is 1.8 which is used on the CI systems as well as during the release."

Anyways, I made my case. If you guys see no merit in what I am saying, go ahead and merge.